gateway/gateway.go

```diff
@@ -656,7 +657,27 @@ func (gateway *HandleT) printStats(ctx context.Context) {
 	}
 }
 
-func (gateway *HandleT) stat(wrappedFunc func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+func (gateway *HandleT) withMiddleware(fu func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+	//this is to make sure that we don't have more than `maxClient` in-memory at any point of time. As, having more http client than `maxClient`
+	// may lead to gateway OOM kill.
+	//NOTE: even after we hit the limit of `maxNoCLient`. New request will still pileup until we hit the backlog limit set by `listen` command.
```

Comment:
> backlog limit set by `listen` command

please explain

Reply:
changed the implementation. So, this isn't relevant anymore.